### DISCUSSION TITLE
I am using nextjs and I am wondering, the best way to detect organization switching

### DISCUSSION THREAD

**Discussion by mainawycliffe** [clerk/javascript#3435] [Support request - DO NOT OPEN HERE]
I want to be able to clear the cache when the user switches to a different organization, and I am wondering what is the best way to detect this. It would be nice if we had a callback or something we can subscribe to.

**Comment by wobsoriano** [comment]
Have you checked the [`useOrganization`](https://clerk.com/docs/references/react/use-organization#use-organization-returns) hook?  It has an `organization` object that changes when you switch to a different one.

**Reply by mainawycliffe** [reply]
Does it only get triggered when an organization changes, or can this be triggered if something else changes, like a new member is added?

**Reply by wobsoriano** [reply]
You mean when someone accepted an invitation to an org? I think the answer is no

**Reply by mainawycliffe** [reply]
Not accept an invitation, but when someone switches to a different organization, which they are already members of? I just think, it would be nice if we had a sort of callback for when someone switches org to do some clean up like clearing the caching. For now, I had to use the`afterSelectOrganizationUrl` to a page that clear cache before sending someone back home. 

https://clerk.com/docs/components/organization/organization-switcher

**Reply by wobsoriano** [reply]
Switching between organizations will trigger updates
